cnd/src/network/comit_ln.rs

```diff
@@ -577,3 +577,281 @@ impl NetworkBehaviourEventProcess<oneshot_behaviour::OutEvent<finalize::Message>
+            .enable_all()
+            .threaded_scheduler()
+            .thread_stack_size(1024 * 1024 * 8) // the default is 2MB but that causes a segfault for some reason
+            .build()
+            .unwrap();
+
+        let (alice_key_pair, alice_peer_id, alice_seed) = {
+            let seed = RootSeed::new_random(thread_rng()).unwrap();
+            let key_pair = derive_key_pair(&seed);
+            let peer_id = PeerId::from(key_pair.clone().public());
+            (key_pair, peer_id, seed)
+        };
+
+        let (bob_key_pair, bob_peer_id, bob_seed) = {
+            let seed = RootSeed::new_random(thread_rng()).unwrap();
+            let key_pair = derive_key_pair(&seed);
+            let peer_id = PeerId::from(key_pair.clone().public());
+            (key_pair, peer_id, seed)
+        };
+
+        let mut alice_swarm = SwarmBuilder::new(
+            transport::build_comit_transport(alice_key_pair).unwrap(),
+            ComitLN::new(alice_seed),
+            alice_peer_id.clone(),
+        )
+        .executor(Box::new(TokioExecutor {
+            handle: runtime.handle().clone(),
+        }))
+        .build();
+
+        let mut bob_swarm = SwarmBuilder::new(
+            transport::build_comit_transport(bob_key_pair).unwrap(),
+            ComitLN::new(bob_seed),
+            bob_peer_id.clone(),
+        )
+        .executor(Box::new(TokioExecutor {
+            handle: runtime.handle().clone(),
+        }))
+        .build();
+
+        let bob_addr: Multiaddr = "/ip4/127.0.0.1/tcp/3000".parse().unwrap();
+        Swarm::listen_on(&mut bob_swarm, bob_addr.clone())
+            .with_context(|| format!("Address is not supported: {:?}", bob_addr))
+            .unwrap();
+
+        let send_swap_digest = random_swap_digest();
+
+        let dial_info = DialInformation {
+            peer_id: bob_peer_id.clone(),
+            address_hint: Some(bob_addr.clone()),
+        };
+
+        let ethereum_id = EthereumIdentity::from(Address::default());
+
+        let secp_pubkey = secp256k1::PublicKey::from_str(
+            "02c2a8efce029526d364c2cf39d89e3cdda05e5df7b2cbfc098b4e3d02b70b5275",
+        )
+        .unwrap();
+        let lightning_id = crate::lightning::PublicKey::from(secp_pubkey);
+
+        let swap_params_alice = HanEtherereumHalightBitcoinCreateSwapParams {
+            role: Role::Alice,
+            peer: dial_info.clone(),
+            ethereum_identity: ethereum_id,
+            ethereum_absolute_expiry: Timestamp::from(10),
+            ethereum_amount: asset::Ether::zero(),
+            lightning_identity: lightning_id,
+            lightning_cltv_expiry: Timestamp::from(10),
+            lightning_amount: asset::Lightning::from_sat(0),
+        };
+
+        let swap_params_bob = HanEtherereumHalightBitcoinCreateSwapParams {
+            role: Role::Bob,
+            peer: dial_info, /* not relevant for Bob, should be optional or somehow related
+                              * to role */
```

Comment:
TIL; This comment is not true, the dial info of alice is passed in by Bob since he learned it during the negotiation phase.

Reply:
Yeah, eventually Bob could check if an announce connection opened by Alice was actually opened by Alice' peer-id that he learned during negotiation. 

Thanks @tcharding